File: app/src/main/java/com/hhy/game/dodgebox/SettingsActivity.java
package com.hhy.game.dodgebox;

import android.content.Intent;
import android.content.SharedPreferences;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.view.WindowManager;
import android.widget.ImageButton;
import java.util.ArrayList;

public class SettingsActivity extends AppCompatActivity implements View.OnClickListener {
    // imagebuttons & lists
    private ArrayList<ImageButton> backgrounds = new ArrayList<>();
    private ArrayList<ImageButton> characters = new ArrayList<>();
    ImageButton background1, background2, background3, background4;
    ImageButton character1, character2;
    ImageButton backButton;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        // for fullscreen
        getWindow().setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN, WindowManager.LayoutParams.FLAG_FULLSCREEN);

        setContentView(R.layout.activity_settings);

        // imagebuttons
        background1 = findViewById(R.id.background1);
        background1.setOnClickListener(this);
        background2 = findViewById(R.id.background2);
        background2.setOnClickListener(this);
        background3 = findViewById(R.id.background3);
        background3.setOnClickListener(this);
        background4 = findViewById(R.id.background4);
        background4.setOnClickListener(this);
        character1 = findViewById(R.id.character1);
        character1.setOnClickListener(this);
        character2 = findViewById(R.id.character2);
        character2.setOnClickListener(this);
        backButton = findViewById(R.id.backButton);
        backButton.setOnClickListener(this);

        // image buttons lists
        backgrounds.add(background1);
        backgrounds.add(background2);
        backgrounds.add(background3);
        backgrounds.add(background4);
        characters.add(character1);
        characters.add(character2);

        // load for old chosens
        borderBackground(loadBackground());
        borderCharacter(loadCharacter());

        // for keeping screen on
        getWindow().addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON);
    }


    @Override
    public void onClick(View view) {
        switch (view.getId()) {
            case R.id.background1:
                borderBackground(0);
                saveBackground(0);
                break;
            case R.id.background2:
                borderBackground(1);
                saveBackground(1);
                break;
            case R.id.background3:
                borderBackground(2);
                saveBackground(2);
                break;
            case R.id.background4:
                borderBackground(3);
                saveBackground(3);
                break;
            case R.id.character1:
                borderCharacter(0);
                saveCharacter(0);
                break;
            case R.id.character2:
                borderCharacter(1);
                saveCharacter(1);
                break;

            // backbutton
            case R.id.backButton:
                Intent intent = new Intent(getApplicationContext(), MenuActivity.class);
                startActivity(intent);
                break;
            default:
                break;
        }

    }

    // chosen background
    private void borderBackground(int id) {
        for (ImageButton background : backgrounds) {
            background.setBackgroundResource(0);
        }
        backgrounds.get(id).setBackgroundResource(R.drawable.border);
    }

    // chosen character
    private void borderCharacter(int id) {
        for (ImageButton character : characters) {
            character.setBackgroundResource(0);
        }
        characters.get(id).setBackgroundResource(R.drawable.border);
    }

    // save background
    private void saveBackground(int id) {
        SharedPreferences sharedPreferences = getApplicationContext().getSharedPreferences("dodgebox", MODE_PRIVATE);
        SharedPreferences.Editor editor = sharedPreferences.edit();
        editor.putInt("background", id);
        editor.apply();
    }

    // load last chosen background
    private int loadBackground() {
        SharedPreferences sharedPreferences = getApplicationContext().getSharedPreferences("dodgebox", MODE_PRIVATE);
        return sharedPreferences.getInt("background", 0);
    }

    // save character
    private void saveCharacter(int id) {
        SharedPreferences sharedPreferences = getApplicationContext().getSharedPreferences("dodgebox", MODE_PRIVATE);
        SharedPreferences.Editor editor = sharedPreferences.edit();
        editor.putInt("character", id);
        editor.apply();
    }

    // load last chosen character
    private int loadCharacter() {
        SharedPreferences sharedPreferences = getApplicationContext().getSharedPreferences("dodgebox", MODE_PRIVATE);
        return sharedPreferences.getInt("character", 0);
    }

    @Override
    // on back pressed
    public void onBackPressed() {
        Intent intent = new Intent(getApplicationContext(), MenuActivity.class);
        startActivity(intent);
    }
}

File: app/src/main/java/com/hhy/game/dodgebox/SettingsActivity.kt
package com.hhy.game.dodgebox

import android.app.Activity
import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.view.View
import android.view.WindowManager
import android.widget.ImageButton

class SettingsActivity : Activity(), View.OnClickListener {
    // imagebuttons & lists
    private val backgrounds = ArrayList<ImageButton>()
    private val characters = ArrayList<ImageButton>()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        // for fullscreen
        window.setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN, WindowManager.LayoutParams.FLAG_FULLSCREEN)
        setContentView(R.layout.activity_settings)

        // imagebuttons
        findViewById<ImageButton>(R.id.background1).also {
            it.setOnClickListener(this)
            backgrounds.add(it)
        }
        findViewById<ImageButton>(R.id.background2).also {
            it.setOnClickListener(this)
            backgrounds.add(it)
        }
        findViewById<ImageButton>(R.id.background3).also {
            it.setOnClickListener(this)
            backgrounds.add(it)
        }
        findViewById<ImageButton>(R.id.background4).also {
            it.setOnClickListener(this)
            backgrounds.add(it)
        }
        findViewById<ImageButton>(R.id.character1).also {
            it.setOnClickListener(this)
            characters.add(it)
        }
        findViewById<ImageButton>(R.id.character2).also {
            it.setOnClickListener(this)
            characters.add(it)
        }
        findViewById<ImageButton>(R.id.backButton).also {
            it.setOnClickListener(this)
        }

        // load for old chosens
        borderBackground(loadBackground())
        borderCharacter(loadCharacter())

        // for keeping screen on
        window.addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON)
    }

    override fun onClick(view: View) {
        when (view.id) {
            R.id.background1 -> saveBackground(0)
            R.id.background2 -> saveBackground(1)
            R.id.background3 -> saveBackground(2)
            R.id.background4 -> saveBackground(3)
            R.id.character1 -> saveCharacter(0)
            R.id.character2 -> saveCharacter(1)
            R.id.backButton -> {
                val intent = Intent(this, MenuActivity::class.java)
                intent.flags = Intent.FLAG_ACTIVITY_NEW_TASK
                startActivity(intent)
            }
            else -> {
            }
        }
    }

    // chosen background
    private fun borderBackground(id: Int) {
        for (background in backgrounds) {
            background.setBackgroundResource(0)
        }
        backgrounds[id].setBackgroundResource(R.drawable.border)
    }

    // chosen character
    private fun borderCharacter(id: Int) {
        for (character in characters) {
            character.setBackgroundResource(0)
        }
        characters[id].setBackgroundResource(R.drawable.border)
    }

    // save background
    private fun saveBackground(id: Int) {
        borderBackground(id)
        val sharedPreferences = getSharedPreferences("dodgebox", Context.MODE_PRIVATE)
        sharedPreferences.edit().putInt("background", id).apply()
    }

    // load last chosen background
    private fun loadBackground(): Int {
        val sharedPreferences = getSharedPreferences("dodgebox", Context.MODE_PRIVATE)
        return sharedPreferences.getInt("background", 0)
    }

    // save character
    private fun saveCharacter(id: Int) {
        borderCharacter(id)
        val sharedPreferences = getSharedPreferences("dodgebox", Context.MODE_PRIVATE)
        sharedPreferences.edit().putInt("character", id).apply()
    }

    // load last chosen character
    private fun loadCharacter(): Int {
        val sharedPreferences = getSharedPreferences("dodgebox", Context.MODE_PRIVATE)
        return sharedPreferences.getInt("character", 0)
    }

    // on back pressed
    override fun onBackPressed() {
        val intent = Intent(this, MenuActivity::class.java)
        intent.flags = Intent.FLAG_ACTIVITY_NEW_TASK
        startActivity(intent)
    }
}